WordPress/openverse-frontend
# Add eslint rules for imports and eslint comments

## Fixes


Fixes #901 by @krysal 
Fixes #855 by @sarayourfriend 

## Description


Adds two new ESLint plugins, `eslint-comments` and `import` and configures them and fixes any errors they surfaced.

The biggest change here is of course the import order linting. I tried to create an ordering scheme that more or less matched what we'd been doing on our own. I'm not sure it's perfect and we can bikeshed a little bit about it in this PR (in particular whether treating assets and components as separate groups makes sense).

Overall just hoping to get some consistency :slightly_smiling_face: 

## Testing Instructions

Checkout the branch and try some of the new rules out. Almost all the import ones are auto-fixable which is really nice. The eslint-comments ones take a little more manual work.

Also review each of the plugins and see if there's any rules in their recommended sets that we want to disable. Likewise, see if there are any excluded from the recommended set that we'd like to use.

Please also let me know if there're any additional comments that would be helpful to add to the rule configurations.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
